### PR TITLE
feat(GenomeAnnotationReader): Reader of NCBI's genome annotetion at nuccore databases

### PR DESCRIPTION
**Pull request by acpguedes**
# Description

I included a new reader to read NCBI's genome annotation. It receive a species name and search the organism against NCBI's nuccore databases and return a array of genome annotation of this species to multiple strains. 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ x] New Loader/Tool
- [ ] Bug fix / Smaller change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ x] I have added a library.json file if a new loader/tool was added
- [ x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ x] I ran `make format; make lint` to appease the lint gods

image

### CODE REVIEW

File: llama_hub/genome/base.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from llama_index.readers.schema.base import Document
+
+class GenomeAnnotationReader(BaseReader):
+    """Genome Reader.
+
+    Read read genome annotation from NCBI.
+
+    """
+
+    def load_data(
```
[the remark this answers is not in the document]
Comment (anoopshrma) [reply llama_hub/genome/base.py:14]: For ref: https://github.com/run-llama/llama-hub/blob/main/llama_hub/whatsapp/base.py#L24

[the remark this answers is not in the document]
Comment (acpguedes) [reply llama_hub/genome/base.py:14]: Of course. Fixed.

File: poetry.lock
Comment (EmanuelCampos) [inline poetry.lock:1]: remove the poetry.lock from the commit please!

File: llama_hub/genome/base.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            self, 
+            species, 
+            email,
+            rettype='gb', 
+            retmode='text'
+    ):
```
Comment (anoopshrma) [inline llama_hub/genome/base.py:20]: You can make it typed by adding the the schema of output here.

```
def load_data(...) -> List[Document]:


Then in exception case  return empty list []

```

**Comment by anoopshrma** [comment]
Seems there's still conflict! Could you take a look at this also @acpguedes 

Thanks!

**Pull request closed without merging**